==== triton/lib/Target/LLVMIR/LLVMIRTranslation.cpp#12 - triton/lib/Target/LLVMIR/LLVMIRTranslation.cpp#13 ====
# action=edit type=text
--- triton/lib/Target/LLVMIR/LLVMIRTranslation.cpp	2023-03-24 10:56:56.000000000 -0700
+++ triton/lib/Target/LLVMIR/LLVMIRTranslation.cpp	2023-04-04 08:10:34.000000000 -0700
@@ -285,7 +286,11 @@
 translateTritonGPUToLLVMIR(llvm::LLVMContext *llvmContext,
                            mlir::ModuleOp module, int computeCapability) {
   mlir::PassManager pm(module->getContext());
-  applyPassManagerCLOptions(pm);
+  mlir::registerPassManagerCLOptions();
+  if (failed(applyPassManagerCLOptions(pm))) {
+    llvm::errs() << "failed to apply pass manager CL options\n";
+    return nullptr;
+  }
   auto printingFlags = mlir::OpPrintingFlags();
   printingFlags.elideLargeElementsAttrs(16);
   pm.enableIRPrinting(
